### DISCUSSION TITLE
Xdist in a K8S pod

### DISCUSSION THREAD

**Discussion by Formartha** [pytest-dev/pytest#8863] [Q&A]
Hi,

Currently we use VMs with libvirt hypervisor as our execution machines (linked to Jenkins slaves).

We would like to use K8S as our testing infra. The only problem is how will Xdist work in a pod? Would it have any downside in terms of performance?

Thanks,
Mor

**Comment by RonnyPfannschmidt** [comment]
xdist it self is simply unaware of the k8 env and will use cpus as avaliable, for better selection ensure pods have multiple cpus avaliable and xdist has psutil avaliable
